pkg/provider/azure_vmss_cache.go

```diff
@@ -92,13 +92,11 @@ func (ss *ScaleSet) newVMSSCache() (*azcache.TimedCache, error) {
 					klog.Warning("failed to get the name of VMSS")
 					continue
 				}
-				if scaleSet.OrchestrationMode == "" || scaleSet.OrchestrationMode == compute.OrchestrationModeUniform {
```

Comment:
this has changed the original vmss flex behavior  cc @zmyzheng

Reply:
We should not save VMSS Flex into the vmssCache.
vmssCache is only used for vmss uniform (although the name is a little missleading)
The VMSS Flex cache is inside ss.flexScaleSet.vmssFlexCache

Reply:
The main reason we need two different caches is because the vm list API is different between Uniform and Flex

Reply:
Thanks for the review. I roll back the change and use flex cache to get all vmss. Can you help review again?